src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/CountBasesInReference.java

```diff
@@ -16,22 +18,29 @@
         programGroup = ReferenceProgramGroup.class
 )
 public class CountBasesInReference extends ReferenceWalker {
+    @ArgumentCollection
+    final public SimpleOutputCollection out = new SimpleOutputCollection();
+
     @VisibleForTesting
     final long[] baseCounts = new long[256];
 
     @Override
     public void apply(ReferenceContext referenceContext, ReadsContext read, FeatureContext featureContext) {
-      baseCounts[referenceContext.getBase()]++;
+        baseCounts[referenceContext.getBase()]++;
     }
 
     @Override
-    public Object onTraversalSuccess(){
+    public Object onTraversalSuccess() {
+        final StringBuilder sb = new StringBuilder();
         for (int i = 0; i < baseCounts.length; i++) {
             final long count = baseCounts[i];
-            if (count > 0){
-                System.out.println((char)i + " : " + count );
+            if (count > 0) {
+                sb.append((char) i).append(" : ").append(count).append("\n");
             }
         }
+        out.writeToOutput(sb.toString().getBytes());
```

Comment:
I think just passing in `sb` should be fine, rather than `sb.toString().getBytes()`, since your output method calls `toString()` internally.